Sources/EventSource/EventParser.swift

```diff
@@ -23,20 +24,37 @@ public struct ServerEventParser: EventParser {
     static let lf: UInt8 = 0x0A
     static let colon: UInt8 = 0x3A
 
-    public func parse(_ data: Data) -> [EVEvent] {
-        // Split message with double newline
-        let rawMessages: [Data]
-        if #available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, visionOS 1.0, *) {
-            rawMessages = data.split(separator: [Self.lf, Self.lf])
-        } else {
-            rawMessages = data.split(by: [Self.lf, Self.lf])
-        }
+    func parse(_ data: Data) -> [EVEvent] {
+        let (separatedMessages, remainingData) = splitBuffer(for: buffer + data)
+        buffer = remainingData
+        return parseBuffer(for: separatedMessages)
+    }
 
+    private func parseBuffer(for rawMessages: [Data]) -> [EVEvent] {
         // Parse data to ServerMessage model
         let messages: [ServerEvent] = rawMessages.compactMap { ServerEvent.parse(from: $0, mode: mode) }
 
         return messages
     }
+
+    private func splitBuffer(for data: Data) -> (completeData: [Data], remainingData: Data) {
+        let separator: [UInt8] = [Self.lf, Self.lf]
+        var rawMessages = [Data]()
+
+        // If event separator is not present do not parse any unfinished messages
+        guard let lastSeparator = data.lastRange(of: separator) else { return ([], data) }
+
+        let bufferRange = data.startIndex..<lastSeparator.upperBound
```

Comment:
Shouldn't this use `lastSeparator.lowerBound`?
```suggestion
        let bufferRange = data.startIndex..<lastSeparator.lowerBound
```

Reply:
It would be removed by `split` but `lowerBound` makes sense, thanks for noticing 🙂